### PR TITLE
Fix task window title

### LINKED ISSUE #218: Updating a task's name doesn't trigger an update of the window title
It seems like that is because https://github.com/enthought/pyface/blob/master/pyface/tasks/task_window.py#L32
only listens to a change of the active task, not a change in its name.

### PR DESCRIPTION
**Pull request by corranwebster**
Fixes #218.  Includes tests.  We now listen to `active_task.name` instead of just `active_task`.

### CODE REVIEW

**Comment by corranwebster** [comment]
@jonathanrocher do you have time to review this?

**Comment by jonathanrocher** [comment]
Yes, I will take a look today.

**Comment by jonathanrocher** [comment]
Functionally, I confirm that this works. Reviewing code for styling now...

File: pyface/tasks/tests/test_task_window.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from traits.testing.unittest_tools import UnittestTools
+from pyface.tasks.api import Task
+from ..task_window import TaskWindow
+
+
+class TestTaskWindow(unittest.TestCase, UnittestTools):
+
+    def test_title(self):
+        task_1 = Task(name='Test Task')
+        task_2 = Task(name='Test Task 2')
```
Comment (jonathanrocher) [inline pyface/tasks/tests/test_task_window.py]: Would move these 2 lines into a `setUp`.

File: pyface/tasks/tests/test_task_window.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        task_window = TaskWindow(tasks=[task_1, task_2])
+
+        # initially nothing
```
Comment (jonathanrocher) [inline pyface/tasks/tests/test_task_window.py]: I would recommend to break each of these sections to a separate test to make them more granular.

File: pyface/tasks/tests/test_task_window.py (hunk 1 above)
Comment (jonathanrocher) [inline pyface/tasks/tests/test_task_window.py]: Actually the definition of `task_window` can also be moved to a `setUp`.

File: pyface/tasks/tests/test_task_window.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        with self.assertTraitChanges(task_window, 'title', count=1):
+            task_window.active_task = None
+        self.assertEqual(task_window.title, "Window title")
+
+        # change window title
+        with self.assertTraitChanges(task_window, 'title', count=1):
+            task_window.title = "New Window title"
+        self.assertEqual(task_window.title, "New Window title")
+
+        # go back to getting title from task
```
Comment (jonathanrocher) [inline pyface/tasks/tests/test_task_window.py]: Again, I would break each of these sections into a separate test.

**Comment by jonathanrocher** [comment]
Only minor recommendations on making the tests more granular. Other than that, LGTM.

**Comment by corranwebster** [comment]
Fixes as suggested - didn't use a setUp since eventually we should have more tests of other aspects which need different set up, but instead moved object creation into a utility function.

**Comment by corranwebster** [comment]
...and appveyor is failing because pip was upgraded, by the look of things.

**Pull request merged**